src/components/kubectl/port-forward.ts

```diff
@@ -105,62 +97,102 @@ export async function portForwardKubernetes(kubectl: Kubectl, explorerNode?: any
-        if (portMapping.length !== 0) {
-            portForwardToPod(kubectl, podSelection, portMapping, namespace);
-        }
+        await promptAndForwardPort(kubectl, 'pods', podSelection, namespace);
+    }
+}
+
+/**
+ * Prompts the user on what port to port-forward to, and sets up the forwarding
+ * if a valid input was provided.
+ */
+async function promptAndForwardPort(kubectl: Kubectl, kind: string, resourceName: string, namespace?: string): Promise<void> {
+    const portMapping = await promptForPort(kubectl, kind, resourceName, namespace);
+    if (portMapping.length !== 0) {
+        portForwardToResource(kubectl, kind, resourceName, portMapping, namespace);
     }
 }
 
 /**
  * Given a JSON representation of a Pod, extract the ports to suggest to the user for
  * for port forwarding.
  */
-function extractPodPorts(podJson: string): string | undefined {
+function extractPodPorts(podJson: string): ExtractedPort[] {
     const pod = JSON.parse(podJson) as kubernetes.V1Pod;
+    const ports = Array.of<ExtractedPort>();
     const containers = pod.spec.containers;
-    const ports = Array.of<number>();
     containers.forEach((container) => {
         if (container.ports) {
-            const containerPorts = container.ports.map((port) => port.containerPort);
+            const containerPorts = container.ports.map(({name, containerPort}) => ({name, port: containerPort}));
             ports.push(...containerPorts);
         }
     });
-    if (ports.length > 0) {
-        const portPairs = ports.map((port) => `${port}:${port}`);
-        return portPairs.join(' ');
-    }
-    return;
+    return ports;
+}
+
+/**
+ * Given a JSON representation of a Service, extract the ports to suggest to the user
+ * for port forwarding.
+ */
+function extractServicePorts(podJson: string): ExtractedPort[] {
+    const service = JSON.parse(podJson) as kubernetes.V1Service;
+    return service.spec.ports;
+}
+
+/**
+ * Given a JSON representation of a Deployment, extract the ports to suggest to the user
+ * for port forwarding.
+ */
+function extractDeploymentPorts(podJson: string): ExtractedPort[] {
+    const deployment = JSON.parse(podJson) as kubernetes.V1Deployment;
+    const ports = Array.of<ExtractedPort>();
+    const containers = deployment.spec.template.spec.containers;
+    containers.forEach((container) => {
+        if (container.ports) {
+            const containerPorts = container.ports.map(({name, containerPort}) => ({name, port: containerPort}));
+            ports.push(...containerPorts);
+        }
+    });
+    return ports;
 }
 
 /**
  * Prompts the user on what port to port-forward to, and validates numeric input.
  * @returns An array of PortMapping objects.
  */
-async function promptForPort(kubectl?: Kubectl, podName?: string, namespace?: string): Promise<PortMapping[]> {
+async function promptForPort(kubectl: Kubectl, kind: string, resourceName?: string, namespace?: string): Promise<PortMapping[]> {
```

Comment:
Optional can be removed from `resourceName` as the calling method doesn't have this as optional and it's the only one which is using `prompForPort`